src/Testing/src/xunit/QuarantinedTestAttribute.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    /// <summary>
+    /// Marks a test as "Quarantined" so that the build will sequester it and ignore failures.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This attribute works by applying xUnit.net "Traits" based on the criteria specified in the attribute
+    /// properties. Once these traits are applied, build scripts can include/exclude tests based on them.
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// [Fact]
+    /// [QuarantinedTest]
+    /// public void FlakyTest()
+    /// {
+    ///     // Flakiness
+    /// }
+    /// </code>
+    ///
+    /// <para>
+    /// The above example generates the following facet:
+    /// </para>
+    ///
+    /// <list type="bullet">
+    /// <item>
+    ///     <description><c>Quarantined</c> = <c>true</c></description>
+    /// </item>
+    /// </list>
+    /// </example>
+    [TraitDiscoverer("Microsoft.AspNetCore.Testing." + nameof(QuarantinedTestTraitDiscoverer), "Microsoft.AspNetCore.Testing")]
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly)]
+    public sealed class QuarantinedTestAttribute : Attribute, ITraitAttribute
+    {
+        /// <summary>
+        /// Gets a URL to a GitHub issue tracking this quarantined test.
+        /// </summary>
+        public string GitHubIssueUrl { get; }
```

Comment:
Should this be renamed to just something generic like reason, since it doesn't really have to be a tracking issue anymore, but its just random optional metadata

Reply:
Yeah, that sounds good. Let's call it `Reason`, but include a note about GitHub URLs in the summary, perhaps: `Gets an optional reason for the quarantining, such as a link to a GitHub issue URL with more details as to why the test is quarantined"